### PR TITLE
Pass correct prefix length to removeCacheVersionAndGen()

### PR DESCRIPTION
**Pull request by hangshao0**
Persistent cache and non-persistent cache have different prefixes, pass
the correct prefix length to removeCacheVersionAndGen(), so that
Non-persistent cache is able to figure out the correct cache name (for a
lower layer) from a cache unique ID.

Signed-off-by: Hang Shao <hangshao@ca.ibm.com>

### CODE REVIEW

File: runtime/shared_common/OSCache.cpp [hunk 1]
```diff
@@ -1313,8 +1314,19 @@ SH_OSCache::getCacheNameAndLayerFromUnqiueID(J9JavaVM* vm, const char* cacheDirN
 	char nameWithVGenCopy[J9SH_MAXPATH];
 	memset(nameWithVGenCopy, 0, J9SH_MAXPATH);
 	strncpy(nameWithVGenCopy, cacheNameWithVGenStart, cacheNameWithVGenEnd - cacheNameWithVGenStart);
+	
+	J9PortShcVersion versionData;
+	getValuesFromShcFilePrefix(PORTLIB, nameWithVGenCopy, &versionData);
+	UDATA prefixLen = J9SH_VERSION_STRING_LEN + 1;
+	if (J9PORT_SHR_CACHE_TYPE_NONPERSISTENT == versionData.cacheType) {
+#if defined(WIN32)
```
Comment (pshipton) [inline runtime/shared_common/OSCache.cpp]: It would be useful to have comments that explain the differences in the multiple cases.

**Comment by pshipton** [comment]
What was broken by this problem?

**Comment by hangshao0** [comment]
> What was broken by this problem?

Non-persistent cache was not able to figure out the correct cache name (for a lower layer) from a cache unique ID.

**Comment by pshipton** [comment]
> Non-persistent cache was not able to figure out the correct cache name (for a lower layer) from a cache unique ID.

Please include that information in the commit comment.

File: runtime/shared_common/OSCache.cpp (hunk 1 above)
Comment (hangshao0) [reply runtime/shared_common/OSCache.cpp]: Comments added

**Comment by hangshao0** [comment]
> Please include that information in the commit comment.

Done.

**Comment by pshipton** [comment]
jenkins test sanity plinux jdk8

**Comment by pshipton** [comment]
jenkins test sanity win jdk8

**Pull request merged**